torch/_inductor/ir.py

```diff
@@ -3622,14 +3622,17 @@ def codegen_kwargs(self):
         if V.graph.cpp_wrapper:
             # FIXME: we should unconditionally fill self.kwargs with missing default values
             # instead of carrying an extra self.ordered_kwargs_for_cpp_kernel
-            if self.kwargs:
+            if self.kwargs and not self.ordered_kwargs_for_cpp_kernel:
                 assert (
                     self.ordered_kwargs_for_cpp_kernel
                 ), "ordered_kwargs_for_cpp_kernel is missing"
             kwargs = []
             for arg_name in self.ordered_kwargs_for_cpp_kernel:
                 v = self.get_kwargs_value(arg_name)
-                kwargs.append(V.graph.wrapper_code.val_to_arg_str(v))
+                if isinstance(v, sympy.Expr):
```

Comment:
Could we add a clause to handle ```sympy.Expr``` from CppWrapperCodeGen's ```val_to_arg_str``` so that we could make kwargs hold only strings.

Reply:
But I think the point here is to use `sympy.Expr` as kwarg to codegen it at a later stage? cc @oulgen.

Reply:
Yes, if we converted to a string, then codegen will not be able to tell it was a sympy expr and emit it like a tensor. @chenyang78 The problem is codegen tries to figure out things based on their string representation but for sympy expr, it keeps them as is.. which is not great.

Reply:
I see. Thanks for the clarification. This is quite unfortunate...